web/src/utils/metrics.ts

```diff
@@ -85,26 +85,30 @@ const parseMetric = (
   const normalized = normalizeMetrics(raw.values, start, end, step);
   const stats = computeStats(normalized);
 
-  const source: TopologyMetricPeer = {
-    addr: raw.metric.SrcAddr,
-    name: raw.metric.SrcK8S_Name,
-    namespace: raw.metric.SrcK8S_Namespace,
-    ownerName: raw.metric.SrcK8S_OwnerName,
-    ownerType: raw.metric.SrcK8S_OwnerType,
-    type: raw.metric.SrcK8S_Type,
-    hostName: raw.metric.SrcK8S_HostName
-  };
-  const destination: TopologyMetricPeer = {
-    addr: raw.metric.DstAddr,
-    name: raw.metric.DstK8S_Name,
-    namespace: raw.metric.DstK8S_Namespace,
-    ownerName: raw.metric.DstK8S_OwnerName,
-    ownerType: raw.metric.DstK8S_OwnerType,
-    type: raw.metric.DstK8S_Type,
-    hostName: raw.metric.DstK8S_HostName
-  };
-  source.displayName = buildPeerDisplayName(source, scope);
-  destination.displayName = buildPeerDisplayName(destination, scope);
+  let source: TopologyMetricPeer | undefined = undefined;
```

Comment:
Wouldn't this have still worked with the previous version, without the condition on scope 'app'? Since all these fields can be undefined. This would avoid having `source` and `destination` being optional (as their content is optional), hence avoid all these `!` non-null assertion spreading everywhere? The non-null assertions can be sometimes useful but most of the time it's a code smell and should be avoided.

Reply:
Yes we can keep source / destination with empty fields if you prefer

Reply:
Just forced pushed to revert that
https://github.com/netobserv/network-observability-console-plugin/commit/13c18198d5dd1d9ecf00d79479a42dbb5c6b6edc